src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventCounter.cs

```diff
@@ -228,31 +255,49 @@ IEnumerator IEnumerable.GetEnumerator()
         #endregion // Implementation of the IEnumerable interface
     }
 
-    internal class EventCounterGroup : IDisposable
+    internal class EventCounterGroup
     {
         private readonly EventSource _eventSource;
-        private readonly int _eventSourceIndex;
         private readonly List<EventCounter> _eventCounters;
 
-        internal EventCounterGroup(EventSource eventSource, int eventSourceIndex)
+        internal EventCounterGroup(EventSource eventSource)
         {
             _eventSource = eventSource;
-            _eventSourceIndex = eventSourceIndex;
             _eventCounters = new List<EventCounter>();
             RegisterCommandCallback();
         }
 
-        private void Add(EventCounter eventCounter)
+        internal void Add(EventCounter eventCounter)
         {
-            _eventCounters.Add(eventCounter);
+            lock (this) // Lock the EventCounterGroup
+                _eventCounters.Add(eventCounter);
+        }
+
+        internal void Remove(EventCounter eventCounter)
+        {
+            lock (this) // Lock the EventCounterGroup
+                _eventCounters.Remove(eventCounter);
         }
 
         #region EventSource Command Processing
 
         private void RegisterCommandCallback()
         {
-#if SUPPORTS_EVENTCOMMANDEXECUTED
+            // Old destktop runtimes don't have this 
+#if !NO_EVENTCOMMANDEXECUTED_SUPPORT
```

Comment:
Nit: Why the double negative rather than defining EVENTCOMMANDEXECUTED_SUPPORT when it's supported?